File: Chapter9ActivityHook1/app/src/main/java/com/mars/activityhook1/classloader_hook/CustomClassLoader.java
package com.mars.activityhook1.classloader_hook;

import android.util.Log;

import dalvik.system.DexClassLoader;

/**
 * Created by geyan.marx on 2020/10/24
 * 自定义的ClassLoader, 用于加载"插件"的资源和代码
 */
public class CustomClassLoader extends DexClassLoader {

    public CustomClassLoader(String dexPath, String optimizedDirectory, String librarySearchPath, ClassLoader parent) {
        super(dexPath, optimizedDirectory, librarySearchPath, parent);
        Log.e("gy", "CustomClassLoader---dexPath = " + dexPath +
                " , optimizedDirectory = " + optimizedDirectory +
                " , librarySearchPath = " + librarySearchPath);
    }
}

File: Chapter9ActivityHook1_V2/app/src/main/java/com/mars/chapter9activityhook1_v2/CustomClassLoader.kt
package com.mars.chapter9activityhook1_v2

import android.util.Log
import dalvik.system.DexClassLoader

/**
 * Created by geyan on 2021/1/17
 */
class CustomClassLoader(
    private val dexPath: String,
    private val optimizedDirectory: String,
    private val librarySearchPath: String,
    private val parentClassLoader: ClassLoader? = null
) : DexClassLoader(dexPath, optimizedDirectory, librarySearchPath, parentClassLoader) {

    init {
        Log.e(
            "gy", "CustomClassLoader---dexPath = $dexPath, " +
                    "optimizedDirectory = $optimizedDirectory, " +
                    "librarySearchPath = $librarySearchPath, " +
                    "parentClassLoader = $parentClassLoader"
        )
    }
}